test_common.c

```diff
@@ -51,6 +51,18 @@ int main(int argc, char **argv)
     } else
         printf("REF mechanism\n");
 
+    char *Top = word;
+    char *pool = NULL;
+
+    if (CPYmask) {
+        /* memory pool */
+        pool = (char *) malloc(poolsize * sizeof(char));
```

Comment:
According to C99 section 6.5.3.4, `sizeof(char)` is always `1`. So, you can rewrite as `malloc(poolsize)`. Also, you don't have to cast to `char *` since the return type of `malloc` is `void *`.